ci/docker.go

```diff
@@ -63,27 +63,41 @@ func (d *Docker) Run(ctx context.Context, job *Job) (string, error) {
 	// wait until the container stops or context times out.
 	_, err = cli.ContainerWait(ctx, resp.ID)
 	if err != nil {
-		if errors.Is(err, context.DeadlineExceeded) {
-			timeout := time.Duration(1 * time.Second)
-			// ContainerStop seems to work better than ContainerKill in my tests.
-			stopErr := cli.ContainerStop(context.Background(), resp.ID, &timeout)
-			if stopErr != nil {
-				// if we failed to stop, report it up the chain
-				err = stopErr
-			}
-			userMsg := "Container timeout. Please check for infinite loops or other slowness."
-			return userMsg, err
+		if !errors.Is(err, context.DeadlineExceeded) {
+			return "", err
 		}
-		return "", err
+
+		// stop runaway container whose deadline was exceeded
+		timeout := time.Duration(1 * time.Second)
+		stopErr := cli.ContainerStop(context.Background(), resp.ID, &timeout)
+		if stopErr != nil {
+			return "", stopErr
+		}
+
+		// remove the docker container (when stopped due to timeout) to prevent too many open files
+		rmErr := cli.ContainerRemove(context.Background(), resp.ID, types.ContainerRemoveOptions{})
+		if rmErr != nil {
+			return "", rmErr
+		}
+
+		// return message to user to be shown in the results log
+		return "Container timeout. Please check for infinite loops or other slowness.", err
 	}
 
+	// extract the logs before removing the container below
 	logReader, err := cli.ContainerLogs(ctx, resp.ID, types.ContainerLogsOptions{
 		ShowStdout: true,
 	})
 	if err != nil {
 		return "", err
 	}
 
+	// remove the container when finished to prevent too many open files
+	err = cli.ContainerRemove(ctx, resp.ID, types.ContainerRemoveOptions{})
```

Comment:
When we tried the `ContainerRemove()` last year we would get no container output. But the docker version was older, so it might work now, just needs to be tested.

Reply:
Yeah, see the log message. That's why I remove it after `cli.ContainerLogs()` which is used to get the logs from the container before removing it.

Reply:
Also, the tests passes locally. Meaning that the output is as expected, even though the image was removed. I confirmed that the image was removed with `docker ps -a`.

Reply:
Oh, right. Then it should be ok.